### ISSUE TITLE
Make system easier

### ISSUE THREAD

**Issue by trytomakeyouprivate** [iaacornus/Fedora-OSTree-Setup#32]
- [ ] autoupdates
- [ ] Link useful folders:
- [ ] Flatpak app folder
- [ ] firefox folder `~/.mozilla/firefox/`
- [ ] maybe Brave config folder `~/.var/app/com.brave.Browser/config/BraveSoftware/Brave-Browser/`
- [ ] Libreoffice config folder `~/.var/app/org.libreoffice.LibreOffice/config/libreoffice`
- [ ] Autostart folder `~/.config/autostart`
- [ ] add .hidden file to $HOME
- [ ] autoupdate Arkenfox user.js

**Comment by trytomakeyouprivate** [comment]
```
printf """R
Monero
crash-waydroid-journalctl.txt
hs_err_pid2.log
lynis-report.dat
lynis.log
testdisk.log
qphotorec.log""" > ~/.hidden
```

if you have files to add, add them here

**Comment by trytomakeyouprivate** [comment]
what folder should be use for that? a direct "tools" folder? I find it immensly useful to have, especially if you give icons to the folders. This will be Filemanager-specific though I guess.

**Comment by iaacornus** [comment]
> * [x]  autoupdates
> 
>     * [x]  Link useful folders:
> 
>     * [x]  Flatpak app folder
> 
>     * [x]  firefox folder `~/.mozilla/firefox/`
> 
>     * [x]  maybe Brave config folder `~/.var/app/com.brave.Browser/config/BraveSoftware/Brave-Browser/`
> 
>     * [x]  Libreoffice config folder `~/.var/app/org.libreoffice.LibreOffice/config/libreoffice`
> 
>     * [x]  Autostart folder `~/.config/autostart`
> 
>     * [x]  add .hidden file to $HOME
> 
>     * [ ]  autoupdate Arkenfox user.js

i'm not sure what's the monero for, since it does not seem needed by the general user population, as well as arkenfox, most of the user in general population simply just use the rpm firefox without any tweaks or the flatpak firefox. I also don't know what is the autostart folder for as well as the hidden files

**Comment by trytomakeyouprivate** [comment]
You can cleanup old deployments using these commands:

```
sudo ostree admin pin -u <index starting from 0>
sudo rpm-ostree cleanup remove -r
```

this may be automated and adapted to always keep the last backup, but for me I had 5 versions piling up.

At least I would add it to the "Tools" folder as a small script to run.

```
sudo printf """ #!/bin/bash
sudo ostree admin pin -u 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
sudo rpm-ostree cleanup remove -r""" > ~/Tools/Scripts/OSTree-remove-old-deployments.sh

chmod +x ~/Tools/Scripts/OSTree-remove-old-deployments.sh
```

A way to determine the number of old deployments may be needed, if it stops when the number is too high. Keeping more than 1 backup is a huge usability problem though.

The grub issue showing everything as duplicates will be fixed in the future.

**Comment by iaacornus** [comment]
> yes it is a bit bloated but I just included everything. Monero is removed, not needed. Arkenfox is more secure, it will be optional and during the creation generates a new profile, so even if you choose yes, you can just keep it as a second profile.

#12 is now prioritized

> Autostart and other hidden files are useful imo, because you dont always have to Ctrl+H and search for the actually useful folders. I tried to focus on user-facing folders like Autostart, that are useful and allow user input.

yeah although what does the autostart and hidden files are for?

**Comment by iaacornus** [comment]
> You can cleanup old deployments using these commands:
> 
> ```
> sudo ostree admin pin -u <index starting from 0>
> sudo rpm-ostree cleanup remove -r
> ```

I usually pin the latest with `sudo ostree admin pin 0` and clean the rest of my system with `rpm-ostree cleanup -bmpr`, followed by `sudo ostree admin cleanup -v`.

> this may be automated and adapted to always keep the last backup, but for me I had 5 versions piling up.
> 
> At least I would add it to the "Tools" folder as a small script to run.
> 
> ```
> printf """#!/bin/bash
> sudo ostree admin pin -u 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20
> sudo rpm-ostree cleanup remove -r
> """ > ~/Tools/Scripts/OSTree-remove-old-deployments.sh
> 
> chmod +x ~/Tools/Scripts/OSTree-remove-old-deployments.sh
> ```
> 
> A way to determine the number of old deployments may be needed, if it stops when the number is too high. I tried counting `"fedora:fedora"` in the `rpm-ostree status` output with `rpm-ostree status | tr -cd 'fedora:fedora' | wc -c` but it doesnt work.
> 
> Keeping more than 1 backup is a huge usability problem though.
> 
> The grub issue showing everything as duplicates will be fixed in the future.

also, `sudo` is not needed by `rpm-ostree`.

**Comment by trytomakeyouprivate** [comment]
crazy sudo is not needed!

Okay these commands seem better. the first removes the old deployments from the list and the second removes the files?

Should that be automated as an autostart script? Or would that be too invasive?

**Comment by trytomakeyouprivate** [comment]
The autostart folder can hold .desktop icons and scripts. KDE has a GUI for that, so maybe not needed. GNOME too?

**Comment by iaacornus** [comment]
> crazy sudo is not needed!

yes the permission handling is done in polkit (discussion from here coreos/rpm-ostree#745)

> Okay these commands seem better. the first removes the old deployments from the list and the second removes the files?
> 
> Should that be automated as an autostart script? Or would that be too invasive?

i don't think it's a good idea to make it a startup script, since for example in my case, i usually keep some version for backup purposes where i can reboot later, also rpm-ostree automatically removes other image layer, so we don't need to run a cleanup everytime.

**Comment by iaacornus** [comment]
> The autostart folder can hold .desktop icons and scripts. KDE has a GUI for that, so maybe not needed. GNOME too?

yeah but what would it autostart tho? not sure what scripts, and `.desktop` goes to `$HOME/.local/share/applications`

**Comment by trytomakeyouprivate** [comment]
> > The autostart folder can hold .desktop icons and scripts. KDE has a GUI for that, so maybe not needed. GNOME too?
> 
> yeah but what would it autostart tho? not sure what scripts, and `.desktop` goes to `$HOME/.local/share/applications`

It autostarts apps or scripts on system boot. For me its:
- nextcloud
- cryptomator
- a Libreoffice table for my timetable
- a script setting my mic volume because its always too high
- a .desktop to enable journalctl logs in a Konsole window
- SyncThingy
- a script removing old backup files

**Comment by iaacornus** [comment]
> > > The autostart folder can hold .desktop icons and scripts. KDE has a GUI for that, so maybe not needed. GNOME too?
> > 
> > 
> > yeah but what would it autostart tho? not sure what scripts, and `.desktop` goes to `$HOME/.local/share/applications`
> 
> It autostarts apps or scripts on ~system boot~ login. For me its:
> 
>     * nextcloud
> 
>     * cryptomator
> 
>     * a Libreoffice table for my timetable
> 
>     * a script setting my mic volume because its always too high
> 
>     * a .desktop to enable journalctl logs in a Konsole window
> 
>     * SyncThingy
> 
>     * a script removing old backup files

that seems rather a personal modification rather than anyone would use for me. also what are these symlinks for?

```
ln -s ~/.mozilla/firefox/ Firefox
ln -s ~/.var/app/org.mozilla.firefox/.mozilla/firefox Firefox-Flatpak
ln -s ~/.var/app/com.brave.Browser/config/BraveSoftware/Brave-Browser/ Brave-Browser
ln -s ~/.var/app/org.libreoffice.LibreOffice/config/libreoffice Libreoffice
ln -s ~/.config/autostart Autostart
ln -s ~/.config Configuration
ln -s ~/.scripts User-Scripts
ln -s ~/.local/share/applications Appstarter
ln -s ~/.local/share/fonts Flatpak-Fonts
ln -s ~/.local/share/Cryptomator/mnt/ Cryptomator-decrypted
ln -s ~/.var/app Flatpak-Apps
ln -s ~/.config/user-dirs.dirs User-Directories

mkdir KDE
cd KDE
ln -s ~/.themes Themes
ln -s ~/.config/kde.org Configuration
ln -s ~/.config/kdedefaults KDE-Defaults
ln -s /usr/share/sddm/themes "SDDM-Themes (write only)"
ln -s ~/.config/menus Appmenu
ln -s ~/.config/discoverrc Discover-Settings
ln -s ~/.config/kdeglobals KDEglobals
ln -s ~/.config/kded_device_automounterrc Automount-Config
ln -s ~/.config/konsolesshconfig KonsoleSSH-Config
ln -s ~/.config/krunnerrc Krunner-Config
ln -s ~/.config/kservicemenurc Servicemenu-Config
ln -s ~/.config/kwinrc Kwin-Config
ln -s ~/.config/mimeapps.list Mimeapps
ln -s ~/.config/okularpartrc Okular-Config
ln -s ~/.config/spectaclerc Spectacle-Config
```

**Comment by trytomakeyouprivate** [comment]
Oh forgot to answer. No autostart is not personal.

It is useful to have the folder there, because through KDE GUI you cant remove some things sometimes. Lots of things profit from autostarting, like Syncthing or Nextcloud, maybe your mail or chat client. You want them there and this is just to make manual configs easier.

The .var folders are for flatpak configs. Dealing with Firefox requires to know where that is, and flatpaks are complicated.

the .config can be removed, but its better than hiding-unhiding your things in /home all the time imo.

the appmenu actually looks understandable, so you could customized it there, and restore it from a previous version. So this is also useful for backup purposes. For example you backup all your data but dont have the apps installed. This menu will look weird then or even reset. After installing all the apps, you need to paste it there again.

discover settings are really easy to disable/enable Fedora Flatpaks and other stuff.

all the other folders are all selected for files that are readable and may be useful.

cryptomator-decrypted is very personal, but I dont know a (not overcomplicated like the json file for applications) way to select what folders to link. For example you will probably not need Firefox Flatpak, RPM and Brave folders, but you can just delete them there, thats what the message in the folders says.

the autostart autostarter is not needed on Silverblue and Kinoite.

**Comment by trytomakeyouprivate** [comment]
print """ #!/bin/bash
# How to re-enable the power-profiles switcher if its gone?

sudo systemctl unmask power-profiles-daemon.service
sudo systemctl restart power-profiles-daemon.service

# Comment: Its said that this may conflict with TLP, but many use both together.
""" > ~/Tools/Scripts/reenable-power-profiles.sh

**Comment by trytomakeyouprivate** [comment]
2Templates: somehow Dolphin uses a completely different folder for the "create new" menu. Without the native Libreoffice and the additional package, there is no "create writer document" entry e.g.

also writing markdown is annoying as there is no element.

All users have to do is change the .desktop extension to .txt and files are no longer executed

Dolphin extension documentation is nearly nonexistant, I try to find what I can and already converted 2 addons to work with Flatpaks. Containers are more complicated though. Also executing complex ffmpeg queries e.g. inside a flatpak container is something that doesnt work for me at the moment.

Templates for "create new" menu:
```
mkdir -p ~/.local/share/templates
wget ... -p ~/.local/share/templates
```

**Comment by trytomakeyouprivate** [comment]
edited list of linked folders, focussing also on embracing users to create dolphin service menus more easily

**Comment by trytomakeyouprivate** [comment]
warning: lit

```
printf """#!/bin/bash
pushd /tmp

author="$(echo $2 | cut -d '/' -f1)"
reponame="$(echo $2 | cut -d '/' -f2)"

if [ ! $3 ]; then
 releasever="$(rpm -E %fedora)"
else
 releasever=$3
fi

if [[ "$1" == "enable" ]]; then
 echo "$author/$reponame -> $releasever"
 curl -fsSL https://copr.fedorainfracloud.org/coprs/$author/$reponame/repo/fedora-$releasever/$author-$reponame-fedora-.repo | sudo tee /etc/yum.repos.d/$author-$reponame.repo
elif [[ "$1" == "remove" ]]; then
 sudo rm /etc/yum.repos.d/$author-$reponame.repo
fi
""" | sudo tee /var/usrlocal/bin/copr

sudo chmod +x /var/usrlocal/bin/copr
```

Now you can just use `copr enable name/repo` normally!

**Comment by iaacornus** [comment]
> ```
> print """ #!/bin/bash
> # How to re-enable the power-profiles switcher if its gone?
> 
> sudo systemctl unmask power-profiles-daemon.service
> sudo systemctl restart power-profiles-daemon.service
> 
> # Comment: Its said that this may conflict with TLP, but many use both together.
> """ > ~/Tools/Scripts/reenable-power-profiles.sh
> ```
> 
> I had a problem where this was gone. Easy solution but not easy to find.

i'm not sure why would `power-profiles` daemon would even be masked in the first place

**Comment by trytomakeyouprivate** [comment]
Hmm, I like the GUI integration to switch between power modes... have no problem using both I think?

**Comment by trytomakeyouprivate** [comment]
Found out how to add new entries ("templates") to the "create new..." menu in Dolphin,
would be really nice to add them seperately when installing Libreoffice, Krita, Inkscape, Gimp, KDEnlive e.g. as they dont have their own things.

They are just desktop files placed in `~/.local/share/templates`, but they link to specific dirs so idk if they work directly

https://github.com/gmm96/KDE-5-Template-Manager/issues/15

I got a lot working now, perfect addition to Flatpak Libreoffice (most common example) or any other template you use often (create a .desktop preset file for anything)

**Comment by trytomakeyouprivate** [comment]
Some emojis where licensed, [here is a freely accessable one](https://github.com/13rac1/twemoji-color-font/releases/latest)

There is a tool to automatically download the latest Github release of something. We would need the Linux tar.gz, for the current version the command could be:

AppleEmojis (licensed)
```
wget https://github.com/samuelngs/apple-emoji-linux/releases/latest/download/AppleColorEmoji.ttf -P ~/.local/share/fonts
# update font cache
fc-cache -f -v
```

Twemoji
```
wget https://github.com/13rac1/twemoji-color-font/releases/download/v14.0.2/TwitterColorEmoji-SVGinOT-Linux-14.0.2.tar.gz -P ~/.local/share/fonts
# unpack and delete archive
tar -xfv ~/.local/share/fonts/TwitterColorEmoji-SVGinOT-Linux-14.0.2.tar.gz
rm ~/.local/share/fonts/TwitterColorEmoji-SVGinOT-Linux-14.0.2.tar.gz
# update font cache
fc-cache -f -v
```

**Comment by trytomakeyouprivate** [comment]
> ```
> ln -s ~/.var/app/org.mozilla.firefox/.mozilla/firefox Firefox-Flatpak
> ln -s ~/.var/app/com.brave.Browser/config/BraveSoftware/Brave-Browser/ Brave-Browser
> ln -s ~/.var/app/org.libreoffice.LibreOffice/config/libreoffice Libreoffice
> ```
> 
> given that these dirs are hard to find, and people barely need to modify it by themselves, i dont really see a point of symlinking it that it would be exposed.
> 
> > the autostart autostarter is not needed on Silverblue and Kinoite.
> 
> i mean the scope of this program is only for ostree systems right, which is only effectively silverblue and kinoite
> 
> > the .config can be removed, but its better than hiding-unhiding your things in /home all the time imo.
> 
> well the thing is it would cluter the home dir, and layman really dont need t omodify and access `.config` folder everytime
> 
> > It is useful to have the folder there, because through KDE GUI you cant remove some things sometimes. Lots of things profit from autostarting, like Syncthing or Nextcloud, maybe your mail or chat client. You want them there and this is just to make manual configs easier.
> 
> this seems a good idea now, thanks for clarifying

I move and delete Firefox stuff A LOT. Editing the UserChrome.css or the user.js, backing up the content to an external drive e.g. its useful, just disussable if we want the efford to make it manual and choose only one browser folder instead of  (Firefox, Firefox Flatpak, Librewolf, Brave, Tor).

We have to see, this script is for regular users, but you dont have to click on a folder just because its there lol.

it just adds a single "Tools" folder to the home dir. In this folder there 
- links to flatpak content (there are dozens use cases, adding Libreoffice Addons, backing up FF, changing configs, removing leftover flatpak folders)
- opening the Cryptomator mounted unencrypted content
- viewing .autostart entries
- a selected set of KDE configs that are editable and easy to read. Really only files meant to be looked at and configured, again you are not forced to use them and its all userspace
- Waydroid folders, you need a root file manager atm, maybe I add a .desktop file opening dolphin as root with that folder
- hidden dolphin features: inside KDE, templates, servicemenus, servicemenu download. Currently this is a complete mess with bad documentation and this makes it easy to see and use them.
- appstarters: you can manually copy-paste them to GUI-fy commands. Really useful to make your workflows better. Example: I have a Captive-portal-Firefox-Appstarter I open when I am in public wifis, it deativates my VPN, opens a firefox profile with a http-only site where I log in and activates my vpn again when closing FF. All through GUI!

Advanced folders:
- lots more, just nice to have. There may be even more useful files and folders to find and they are totally opt-in, but really nice to know:
- DNF repos
- fail2ban config
- firmware updater config
- Font config
- network manager config
- ostree config
- pam
- pulseaudio
- samba
- selinux
- ssh
- systemd
- upower
- usrlocal/bin
- xdg
- var/lib
- bluetooth
- cups
- dnsmasq
- fstab
- geoclue
- mimetypes
- rpm-ostreed
- rsync
- tlp
- userdirs
- whois

very useful

**Comment by trytomakeyouprivate** [comment]
> > ```
> > print """ #!/bin/bash
> > \# How to re-enable the power-profiles switcher if its gone?
> > 
> > sudo systemctl unmask power-profiles-daemon.service
> > sudo systemctl restart power-profiles-daemon.service
> > 
> > # Comment: Its said that this may conflict with TLP, but many use both together.
> > """ > ~/Tools/Scripts/reenable-power-profiles.sh
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I had a problem where this was gone. Easy solution but not easy to find.
> 
> i'm not sure why would `power-profiles` daemon would even be masked in the first place
> 
> > # Comment: Its said that this may conflict with TLP, but many use both together.
> > """ > ~/Tools/Scripts/reenable-power-profiles.sh
> 
> its not recommended, hence it dont suggest using or installing both, if one should be installed one should be removed all together. But replacing power-profiles with tlp sounds good to me, since power-profiles doesnt always perform better, even worse at some times, tracked here #48

using TLP and power-profiles-daemon was a huge problem for my system, getting painfully slow and unusable. Thanks

**Comment by iaacornus** [comment]
> > > ```
> > > print """ #!/bin/bash
> > > \# How to re-enable the power-profiles switcher if its gone?
> > > 
> > > sudo systemctl unmask power-profiles-daemon.service
> > > sudo systemctl restart power-profiles-daemon.service
> > > 
> > > # Comment: Its said that this may conflict with TLP, but many use both together.
> > > """ > ~/Tools/Scripts/reenable-power-profiles.sh
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I had a problem where this was gone. Easy solution but not easy to find.
> > 
> > 
> > i'm not sure why would `power-profiles` daemon would even be masked in the first place
> > > # Comment: Its said that this may conflict with TLP, but many use both together.
> > > """ > ~/Tools/Scripts/reenable-power-profiles.sh
> > 
> > 
> > its not recommended, hence it dont suggest using or installing both, if one should be installed one should be removed all together. But replacing power-profiles with tlp sounds good to me, since power-profiles doesnt always perform better, even worse at some times, tracked here #48
> 
> using TLP and power-profiles-daemon was a huge problem for my system, getting painfully slow and unusable. Thanks

yep, the two conflicts with each other, so one should be removed, i will make an option for user which one to keep, and the one that will be kept will be optimized

**Comment by trytomakeyouprivate** [comment]
I have a new laptop where tlp kills usb and probably HDMI too. But power-profiles also doesnt seem to work, really weird, AMD and all.

RyzenAdj also doesnt work on OSTree systems it seems, as this literally modifies the kernel or something.

So tlp or power-profiles has to be optional. I would preinstall tlp anyways as it doesnt hurt, but well.

```
#TLP:

sudo systemctl disable --now power-profiles-daemon
sudo systemctl mask power-profiles-daemon
sudo systemctl unmask tlp
sudo systemctl enable --now tlp
sudo nano /etc/tlp.d/00-Template.conf

# Power-profiles

sudo systemctl disable --now tlp
sudo systemctl mask tlp
sudo systemctl unmask power-profiles-daemon
sudo systemctl enable --now power-profiles-daemon
```

or something like that. I would add a custom preset file for tlp to that conf, its empty currently but docs are mentioned in the file, there is a wikipage.
